docs/topics/streamfield.md

```diff
@@ -494,6 +494,22 @@ class EventBlock(blocks.StructBlock):
 
 In this example, the variable `is_happening_today` will be made available within the block template. The `parent_context` keyword argument is available when the block is rendered through an `{% include_block %}` tag, and is a dict of variables passed from the calling template.
 
+Similarly, a `get_template` method can be defined to dynamically select a template based on the block value:
+
+```python
+import datetime
+
+class EventBlock(blocks.StructBlock):
+    title = blocks.CharBlock()
+    date = blocks.DateBlock()
+
+    def get_template(self, value, context=None):
+        if value['date'] > datetime.date.today():
+            return 'myapp/blocks/future_event.html'
+        else:
+            return 'myapp/blocks/event.html'
```

Comment:
Just in case we'd like to use something like [blacken-docs](https://github.com/adamchainz/blacken-docs) in the future.

```suggestion
    def get_template(self, value, context=None):
        if value["date"] > datetime.date.today():
            return "myapp/blocks/future_event.html"
        else:
            return "myapp/blocks/event.html"
```

Also, is not having `=None` for the `value` parameter intentional? Perhaps to conform with `render()`, especially once we drop the deprecation for those who don't have `value` in the signature? If so, then ignore this comment

Reply:
Sounds good... formatting updated in e9e6eaed5bf7435d6288b2961267b2c6c10420c7. (The existing code snippets liberally use single quotes, but I guess anyone who's bothered about consistency will be just as happy to see Black formatting here :-) )

I went with this signature to match the `get_context` example above - I think it's worth trying to present these consistently, and hide the internal "warts" of the deprecation as much as possible from developers. I'm not overly attached to it if there are good reasons for not doing it that way, though...